test/tests.py

```diff
@@ -0,0 +1,142 @@
+from operations import create_index, check_index, create_document,\
+    delete_document, delete_index, get_document
+from http import HTTPStatus
+from typing import Tuple, Callable
+import unittest
+import os
+import logging
+import time
+import requests
+
+logger = logging.getLogger(__name__)
+
+
+# The following "retry_request" function's purpose is to retry a certain request for "max_attempts"
+# times every "delay" seconds IF the requests returned a status code other than what's expected.
+# So this "retry_request" function's arguments are a request function's name and whatever arguments that function
+# expects, and the status code the request function is expecting to get.
+def retry_request(request: Callable, args: Tuple = (), max_attempts: int = 10, delay: float = 0.5,
+                  expected_status_code: HTTPStatus = None):
+    for attempt in range(1, max_attempts + 1):
+
+        result = request(*args)
```

Comment:
Where is the exception we discussed? Based on the demo conversation I thought you were going to add this back in and handle connection refused errors?

Reply:
I spoke with Mikayla after the demo conversation, we agreed that there are two valid point of views here:
- Adding the exception handling + make the first time each endpoint has a request made to retry-able.
- Acknowledge that the exception handling we discussed would not be necessary as the code would not be reached if the prerequisites (making sure all the containers are up) for the test script are followed properly.

I pushed the commit I had prepared which follows the second point of view, but later realized that it can still be helpful in cases such as the containers are up but ports are not set properly, therefore it can still be helpful to the user of this test. Pushing the commit with the exception handling now. It will handle all cases where any of the containers are down.